nix-mode.el

```diff
@@ -353,27 +373,74 @@ STRING-TYPE type of string based off of Emacs syntax table types"
     (when matching-indentation (indent-line-to matching-indentation) t))
   )
 
-(defun nix-indent-find-BOL-expression-start ()
-  (beginning-of-line)
-  (let ((counter 1))
-    (while (and (> counter 0) (re-search-backward "\\(;\\|=\\|inherit\\|with\\b\\)" nil t))
-      (unless (nix--inside-string-or-comment)
-        (setq counter (cond ((looking-at "with\\|=\\|inherit") (- counter 1))
-                            ((looking-at ";") (+ counter 1))))
-        )
-      )
-      (when (/= counter 0) (goto-char (point-min))) t))
+(defun nix-mode-make-regexp (parts)
+  "Combine the regexps into a single or-delimited regexp."
+  (declare (indent defun))
```

Comment:
What does this one do?

Reply:
`nix-mode-make-regexp` takes a list of regexps and makes a larger regex out of them separated by `\\|` so that any of the constituents match.